src/sentry/models/groupsearchview.py

```diff
@@ -0,0 +1,43 @@
+from django.db import models
+from django.db.models import UniqueConstraint
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import region_silo_model
+from sentry.db.models.base import DefaultFieldsModel
+from sentry.db.models.fields.foreignkey import FlexibleForeignKey
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+from sentry.models.savedsearch import SortOptions
+
+
+@region_silo_model
+class GroupSearchView(DefaultFieldsModel):
+    """
+    A model for a user's view of the issue stream
+    """
+
+    __relocation_scope__ = RelocationScope.Organization
+    name = models.TextField(max_length=128)
+    user_id = HybridCloudForeignKey("sentry.User", on_delete="CASCADE")
+    organization = FlexibleForeignKey("sentry.Organization")
+
+    query = models.TextField()
+    query_sort = models.CharField(
+        max_length=16, default=SortOptions.DATE, choices=SortOptions.as_choices()
+    )
+    position = models.PositiveSmallIntegerField()
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_groupsearchview"
+        unique_together = ()
+        # Two views cannot occupy the same position in an organization user's list of views
+        constraints = [
+            UniqueConstraint(
+                fields=["user_id", "organization_id", "position"],
```

Comment:
Btw just a little gotcha with this:

Just say you have rows like
```
id, user_id, org_id, position
1, 100, 1000, 0
2, 100, 1000, 1
```
Now let's say you want to switch the position of these rows. When you attempt the update, you'll run into the constraint blocking you, since you're do `update ... where id = 1 set position = 1`, `update ... where id = 2 set position = 0`, which violates the constraint.

You can mark the constraint deferrable to avoid this https://docs.djangoproject.com/en/5.0/ref/models/constraints/#deferrable

Reply:
This just saved me a huge headache down the line thank you! 🙏  Adding that in now

Reply:
You'll need to do any position swaps in a transaction or postgres will be sad.